src/dune/dune_package.mli

```diff
@@ -67,7 +67,7 @@ module Or_meta : sig
   val pp :
     dune_version:Dune_lang.Syntax.Version.t -> Format.formatter -> t -> unit
 
-  val load : Dpath.t -> t
+  val load : Dpath.t -> (t, [ `Invalid_dune_package of exn ]) Result.t
```

Comment:
We can also use `Or_exn` here, since we use it in several other places for the same purpose, i.e. delaying raising an error

Reply:
I switched to `Or_exn.t`. My original motivation with the polymorphic variant was to signify that only parsing errors would be caught here.